pkg/jobs/hive/hive.go

```diff
@@ -426,5 +430,35 @@ func validateUpgradeVersion(client clientv1.Client, clusterName string, curator
 		return errors.New("Provided version is not valid")
 	}
 
+	isValidChannel := false
+	if channel != "" {
+		for _, versionRelease := range managedClusterInfo.Status.DistributionInfo.OCP.VersionAvailableUpdates {
+			if versionRelease.Version == desiredUpdate {
+				for _, c := range versionRelease.Channels {
+					if c == channel {
+						isValidChannel = true
+						break
+					}
+				}
+			}
+		}
+	}
+	if channel != "" && !isValidChannel {
+		return errors.New("Provided channel is not valid")
+	}
+
+	isValidUpstream := false
+	if channel != "" {
+		for _, versionRelease := range managedClusterInfo.Status.DistributionInfo.OCP.VersionAvailableUpdates {
+			if versionRelease.Version == desiredUpdate && versionRelease.URL == upstream {
```

Comment:
Any reason to not put the versionRelease.URL == upstream between line 439 and 440 above to save some looping?